ExtraDry/ExtraDry.Core/Core/InputFormatAttribute.cs

```diff
@@ -7,18 +7,46 @@ namespace ExtraDry.Core;
 [AttributeUsage(AttributeTargets.Property, AllowMultiple = false)]
 public class InputFormatAttribute : Attribute
 {
+    private PropertySize? size;
+
     /// <summary>
     /// Provides a means to override the default input that is rendered for this property.
     /// </summary>
     public Type? DataTypeOverride { get; set; }
 
     /// <summary>
-    /// The icon to be used in the display of this propertys input, typically on the left of the input
+    /// The icon to be used in the display of this property's input, typically on the left of the input
     /// </summary>
     public string? Icon { get; set; }
 
     /// <summary>
-    /// The affordance icon to be used in the display of this propertys input, typically on the right of the input.
+    /// The affordance icon to be used in the display of this property's input, typically on the right of the input.
     /// </summary>
     public string? Affordance { get; set; }
+
+    /// <summary>
+    /// Gets or sets the size that this field should be displayed.  If this property is not set 
+    /// then the presentation layer will automatically determine the size.  Setting this property
+    /// explicitly allows an override of the default behavior of the presentation layer.
+    /// </summary>
+    /// <remarks>
+    /// Consumers must use the <see cref="GetSize" /> method to retrieve the value, as this
+    /// property getter will throw an exception if the value has not been set.
+    /// </remarks>
+    /// <exception cref="System.InvalidOperationException">
+    /// If the getter of this property is invoked when the value has not been explicitly set using
+    /// the setter.
+    /// </exception>
+    public PropertySize Size {

```

Comment:
Everything else is using a nullable property to identify 'not set'.  While this pattern is more in keeping with many Microsoft attributes, it looks out of place here.  Is there a specific need?

Reply:
It's a CLR restriction. A nullable enum is not a valid attribute parameter type.
So the way I see it, there's 2 ways to work around this, the first is to use the pattern that has been implemented here, using a non-nullable property, with a nullable backing field, and directing consumers to use a GetXXX() function.
The other, which I'd personally prefer would be a non-nullable enum, and an "Undefined" type value in the enum that is the default.

I feel like we should leave as is, if MS is implementing in this pattern, there must be a reason for it. Although, more than happy to change the approach, or even just add further comments?

Reply:
@aakison I've defaulted the value to a new enum value of "unset". 
I also while i was there removed the explicit numeric values on the enum, it is not persisted.
Finally I renamed the property "SizeOverride" to align with the other property "DataTypeOverride".